.github/workflows/ci.yml

```diff
@@ -48,7 +48,7 @@ jobs:
       - run: cargo install cargo-tarpaulin
       # - run: cargo tarpaulin
       - run: cargo tarpaulin --verbose --all-features --workspace --timeout 120 --out Xml --avoid-cfg-tarpaulin
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v5
```

Comment:
While we're at it, we probably want to update this to use the hash as well. I believe this one would be (from [here](https://github.com/codecov/codecov-action/releases/tag/v5.3.1)):
```
codecov/codecov-action@13ce06bfc6bbe3ecf90edbbf1bc32fe5978ca1d3 #v5.3.1
```

Reply:
`codecov` is failing because `v3` uses an old endpoint with a global rate limit, and `v5` needs an organization account with a token.  So I've changed it back to `v3` for now and disabled the `fail_ci_if_error` flag.